### PR TITLE
Generic Importer Trait

### PR DESCRIPTION
**Pull request by conradludgate**
Creates a new trait
```rust
pub trait Importer: IntoIterator<Item = Result<History>> + Sized {
    const NAME: &'static str;
    fn histpath() -> Result<PathBuf>;
    fn parse(path: impl AsRef<Path>) -> Result<Self>;
}
```
and a function `import` which will use the provided `Importer` type to perform a generic import.

It will:
1. Check for a `$HISTFILE` or call `I::histpath()?`
2. Check that the history file is a file (#66)
3. Parse the history file
4. Iterate over the parsed contents, saving them to the DB in batches

TODO: Tests :)

### CODE REVIEW

File: atuin-client/src/import/zsh.rs [hunk 1]
```diff
@@ -1,47 +1,65 @@
...
+        let mut candidates = [".zhistory", ".zsh_history"].iter();
+        loop {
+            match candidates.next() {
+                Some(candidate) => {
+                    let histpath = home_dir.join(candidate);
+                    if histpath.exists() {
+                        break Ok(histpath);
+                    }
+                }
+                None => break Err(eyre!("Could not find history file. try setting $HISTFILE")),
```
Comment (bl-ue) [inline atuin-client/src/import/zsh.rs:46]:
```suggestion
                None => break Err(eyre!("Could not find history file. Try setting $HISTFILE")),
```

File: src/command/import.rs [hunk 2]
```diff
@@ -55,204 +53,90 @@ impl Cmd {
...
 
-    let histpath = if let Ok(p) = histpath {
+    let histpath = if let Ok(p) = env::var("HISTFILE") {
         let histpath = PathBuf::from(p);
 
-        if !histpath.exists() {
+        if !histpath.is_file() {
             return Err(eyre!(
-                "Could not find history file {:?}. try updating $HISTFILE",
+                "Could not find history file {:?}. Try updating $HISTFILE",
```
Comment (bl-ue) [inline src/command/import.rs:84]:
```suggestion
                "Could not find history file {:?}. Try setting $HISTFILE",
```

File: atuin-client/src/import/resh.rs [hunk 3]
```diff
@@ -14,90 +14,56 @@ use super::{count_lines, Importer};
 use crate::history::History;
 
 #[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
```
Comment (ellie) [inline atuin-client/src/import/resh.rs:17]: didn't know about this, awesome :heart_eyes:

**Review by ellie — APPROVED** [review]
Looks great! :rocket: :sparkles:

**Pull request merged**